src/server/rest/v1/service/TransactionService.ts

```diff
@@ -1071,4 +1085,20 @@ export default class TransactionService {
     );
     return transactions;
   }
+
+  private static async getTransactionsCompleted(action: ServerAction, req: Request): Promise<DataResult<Transaction>> {
+    return await TransactionService.getTransactions(req, action, { completedTransactions: true }, [
```

Comment:
Set the query string Status param to 'completed' and call the getTransactions() in which you can provide a IF to add the proper filter (if Status is not provided return all transactions)

Reply:
Done